tests/lib/in_memory/database.py

```diff
@@ -30,10 +30,10 @@ def setup(self, *input_data):
         table_name_to_items = defaultdict(list)
         for item in input_data:
             table_name_to_items[item.__tablename__].append(item)
-        self.tables = {
-            table_name: self.build_table(items)
-            for table_name, items in table_name_to_items.items()
-        }
+        self.tables = defaultdict(Table.empty)
+
+        for table_name, items in table_name_to_items.items():
+            self.tables[table_name] = self.build_table(items)
```

Comment:
This is a reasonable approach, although we no longer raise an error if you try to look up a column on a frame when the column doesn't exist.  I can't see that this matters though.

An alternative would be to pass in a schema.

Reply:
Yeah. I can't (at the moment) see any important errors that we would miss because of that. I think it's probably simpler than having a schema, so let's leave it like this until we find a reason to do otherwise.